mlflow/server/js/src/experiment-tracking/components/ExperimentListView.js

```diff
@@ -86,76 +121,130 @@ export class ExperimentListView extends Component {
     this.updateSelectedExperiment('0', '');
   };
 
-  handleCheck = (checkedKeys) => {
-    if (checkedKeys.length > 0) {
+  // Add a key if it does not exist, remove it if it does
+  // Always keep at least one experiment checked if it is only the active one.
+  handleCheck = (e, key) => {
+    this.setState((prevState, props) => {
+      let { checkedKeys } = prevState;
+      if (e === true && !prevState.checkedKeys.includes(key)) {
+        checkedKeys = [key, ...prevState.checkedKeys];
+      }
+      if (e === false && props.activeExperimentIds.length !== 1) {
+        checkedKeys = prevState.checkedKeys.filter((i) => i !== key);
+      }
+      return { checkedKeys: checkedKeys };
+    }, this.pushExperimentRoute);
+  };
```

Comment:
I propose a slight naming improvement, usually `e` is used as a shorthand for events
```suggestion
  handleCheck = (checked, key) => {
    this.setState((prevState, props) => {
      let { checkedKeys } = prevState;
      if (checked === true && !prevState.checkedKeys.includes(key)) {
        checkedKeys = [key, ...prevState.checkedKeys];
      }
      if (checked === false && props.activeExperimentIds.length !== 1) {
        checkedKeys = prevState.checkedKeys.filter((i) => i !== key);
      }
      return { checkedKeys: checkedKeys };
    }, this.pushExperimentRoute);
  };
```

Reply:
e is the `onChange` event in this case, it's just a bool though. Maybe change to `event` to make it more clear? I'm good either way.

Reply:
This particular checkbox implementation is tricky and callback uses `isChecked` as a first parameter 🙂 
Next, it's passed through in `onChange={(e) => this.handleCheck(e, item.experiment_id)}` line so I guess we can keep the name consistent

![Screen Shot 2023-02-09 at 10 35 50](https://user-images.githubusercontent.com/104438646/217774365-17c46f42-029b-4662-8f4b-e9cc7ff4c090.png)